core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -30,17 +32,26 @@ object Json {
   private val mapper = new ObjectMapper()
 
   /**
-   * Parse a JSON string into a JsonValue if possible. `None` is returned if `input` is not valid JSON.
-   */
+    * Parse a JSON string into a JsonValue if possible. `None` is returned if `input` is not valid JSON.
+    */
   def parseFull(input: String): Option[JsonValue] =
+    try Option(mapper.readTree(input)).map(JsonValue(_))
+    catch { case _: JsonProcessingException => None }
+
+
+  /**
+   * Parse a JSON string into a JsonValue if possible. `None` is returned if `input` is not valid JSON. This method is currently used
+    * to read the already stored invalid ACLs JSON which was persisted using older versions of Kafka (prior to Kafka 1.1.0). KAFKA-6319
+   */
+  def parseBytesIncludingACLs(input: Array[Byte]): Option[JsonValue] =
```

Comment:
Why do we have this here instead of `Acl`?

Reply:
FYI, there's a `tryParseBytes` method that retains the exception.

Reply:
I tried to retain the json parsing related methods in the `Json` class. Let me know if moving this parsing logic to `Acl` would be more legitimate.

Yep, `tryParseBytes` throws the `JsonProcessingException` instead of `None`. I'll update the PR to use `tryParseButes` instead.

Reply:
This logic is only needed by Acl so it's better to keep it contained as a private method.

Reply:
Gotcha, I'll do that